src/main/scala/consul/v1/common/ConsulRequestBasics.scala

```diff
@@ -1,13 +1,16 @@
 package consul.v1.common
 
+import com.ning.http.client.AsyncHttpClientConfig
 import consul.v1.common.Types.DatacenterId
 import play.api.libs.json._
-import play.api.libs.ws.{WS, WSRequest, WSResponse}
-import play.api.Play.current
+import play.api.libs.ws.{WS, WSClient, WSRequest, WSResponse}
+import play.api.libs.ws.ning.NingWSClient
+import play.api.Application
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 
-class ConsulRequestBasics(token: Option[String]) {
+class ConsulRequestBasics(token: Option[String], client: WSClient) {
+  def this(token: Option[String])(implicit app: Application) = this(token, WS.client)
```

Comment:
I don't really want to go with a default argument here, since then we'd need an implicit `Application` for the main constructor. I don't see why @codacy-bot is complaining about this, since it's the normal way to define auxiliary constructors.